src/Chapter1.hs

```diff
@@ -541,8 +543,11 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x <= y && y <= z || z <= y && y <= x = y
+    | y <= x && x <= z || z <= x && x <= y = x
+    | x <= z && z <= y || y <= z && z <= x = z
```

Comment:
That is the correct solution!
You covered all possible cases with these guards 👏🏼 

As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = z`, to tell the compiler, that your pattern matching is exhaustive 🙂
But we can notice that `otherwise = z` is actually __is__ the last case you have in here. Because, if two above are not true, then this one is the only possible option. So no need to check further, the answer will be `z` 🙂